internal/vault/client_factory.go

```diff
@@ -278,8 +300,6 @@ func (m *cachingClientFactory) isDisabled() bool {
 //
 // Supported types for obj are: VaultDynamicSecret, VaultStaticSecret. VaultPKISecret
 func (m *cachingClientFactory) Get(ctx context.Context, client ctrlclient.Client, obj ctrlclient.Object) (Client, error) {
-	m.mu.Lock()
-	defer m.mu.Unlock()
```

Comment:
Just from looking what else uses `mu.Lock()`, any concern with effects between Get() and prune() with respect to cacheKey locking? (i.e. should prune() be using cachKey locks?)

Reply:
Good catch. I've added a read lock in d7d1b288c2c22332543c51175171f1f4ab9f4115